crates/gen-host-wasmtime-rust/src/lib.rs

```diff
@@ -1,31 +1,19 @@
 use heck::*;
-use std::collections::{BTreeMap, BTreeSet, HashMap};
+use std::collections::{BTreeMap, HashMap};
+use std::fmt::Write as _;
 use std::io::{Read, Write};
 use std::mem;
 use std::process::{Command, Stdio};
-use wit_bindgen_core::wit_parser::abi::{AbiVariant, Bindgen, Instruction, LiftLower, WasmType};
-use wit_bindgen_core::{wit_parser::*, Direction, Files, Generator, Source, TypeInfo, Types};
-use wit_bindgen_gen_rust_lib::{
-    to_rust_ident, wasm_type, FnSig, RustFlagsRepr, RustFunctionGenerator, RustGenerator, TypeMode,
+use wit_bindgen_core::wit_parser::abi::AbiVariant;
+use wit_bindgen_core::{
+    uwrite, wit_parser::*, Direction, Files, Generator, Source, TypeInfo, Types,
 };
+use wit_bindgen_gen_rust_lib::{to_rust_ident, FnSig, RustGenerator, TypeMode};
 
 #[derive(Default)]
 pub struct Wasmtime {
     src: Source,
     opts: Opts,
-    needs_get_memory: bool,
-    needs_get_func: bool,
-    needs_char_from_i32: bool,
-    needs_invalid_variant: bool,
-    needs_validate_flags: bool,
-    needs_raw_mem: bool,
-    needs_bad_int: bool,
-    needs_copy_slice: bool,
-    needs_buffer_glue: bool,
-    needs_le: bool,
-    needs_custom_error_to_trap: bool,
-    needs_custom_error_to_types: BTreeSet<String>,
```

Comment:
Could the `Opts::custom_error` field get removed as well?

Also if `tracing` isn't implemented yet mind removing that too?

Reply:
Yes, will get rid of custom error.

Tracing is actually implemented for exports, but not imports - trivial to add it to imports, and configure the tests to turn it on.